docs/developers-guide/substreams-sink-kv.md

```diff
@@ -0,0 +1,231 @@
+---
+description: StreamingFast Substreams sink kv
+---
+
+# `substreams-sink-kv` introduction
+
+## Purpose
+
+This documentation will assist you in using `substreams-sink-kv` to write data from your existing substreams into a key-value store and serve it back through Connect-Web/GRPC.
+
+## Overview
+
+* `substreams-sink-kv` works by reading the output of specially-designed substreams module (usually called `kv_out`) that produces data in a protobuf-encoded structure called `sf.substreams.sink.kv.v1.KVOperations`.
+
+* The data is written to a key-value store (ex: badger, bigtable, TiKV)
+
+* A Connect-Web interface makes the data available directly from the `substreams-sink-kv` process. Alternatively, you can consume the data directly from your key-value store.
+
+## Requirements
+
+* An existing substreams (including substreams.yaml and rust code) that you want to instrument for `substreams-sink-kv`.
+* (Optional) A key-value store where you want to send your data (a badger local file can be used for development or very-small-scale deployments)
+* Knowledge about substreams development (start [here](https://substreams.streamingfast.io/getting-started/quickstart))
+* Rust installation and compiler
+
+## Installation
+
+* Install [substreams-sink-kv](https://github.com/streamingfast/substreams-sink-kv) latest release
+* Install [substreams](https://github.com/streamingfast/substreams) latest release
+* Install [grpcurl](https://github.com/fullstorydev/grpcurl) to easily read the data back from the KV store
+
+## Quick test of your setup
+
+
+* Before you start instrumenting your substreams for the `kv-sink`, it may be a good idea to make sure that you can run your substreams normally:
+
+```bash
+substreams run -e mainnet.eth.streamingfast.io:443 substreams.yaml some_mapper_module --start-block 1000000 --stop-block +1
+```
+
+> **Note** To connect to a public StreamingFast substreams endpoint, you will need an authentication token, follow this [guide](https://substreams.streamingfast.io/reference-and-specs/authentication) to obtain one.
```

Comment:
Useless, we can simply mention that a working Substreams project is expected in `Instrumenting your substreams`